### DISCUSSION TITLE
Which license does transmission follow？GPL or MIT or BSD

### DISCUSSION THREAD

**Discussion by goodpaperman** [transmission/transmission#6836] [Q&A] [answered]
I find 4 licenses under github directory，does transmission follow all of them？GPL conflicts with MIT or BSD，is that possible (to follow different & conflict licenses) ?

**Comment by Pentaphon** [comment]
@ckerr @mikedld perhaps this license issue should be narrowed down to one license? This may dissuade contributors.

**Comment by ckerr** [comment, marked as answer]
Most of the project is dual-licensed under GPLv2 or GPLv3. The boilerplane text is:

>   // This file Copyright © Mnemosyne LLC.
  // It may be used under GPLv2 (SPDX: GPL-2.0-only), GPLv3 (SPDX: GPL-3.0-only),
  // or any future license endorsed by Mnemosyne LLC.
  // License text can be found in the licenses/ folder.

The exceptions to this are some parts that are MIT licensed:

- utp, lpd, dht networking code in libtransmission (as per the contributors' preference)
- macOS client (historical reasons)
- portions of the GTK client (historical reasons)

@Pentaphon narrowing down to a single license is not feasible. The networking code, for example, has to stay MIT. And I'd prefer to keep the majority of the project GPL.

edit: In addition, the `COPYING` file in the root directory spells this out more concisely:

>Copyright 2005-2024. All code is copyrighted by the respective authors.
>
>Transmission can be redistributed and/or modified under the terms of
the GNU GPLv2 (http://www.gnu.org/licenses/license-list.html#GPLv2),
the GNU GPLv3 (http://www.gnu.org/licenses/license-list.html#GNUGPLv3),
or any future license endorsed by Mnemosyne LLC.
>
>In addition, linking to and/or using OpenSSL is allowed.
>
>This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
>
>Some of Transmission's source files have more permissive licenses.
>Those files may, of course, be used on their own under their own terms.
